server/jetstream_cluster.go

```diff
@@ -7691,54 +7691,46 @@ const compressThreshold = 8192 // 8k
 
 // If allowed and contents over the threshold we will compress.
 func encodeStreamMsgAllowCompress(subject, reply string, hdr, msg []byte, lseq uint64, ts int64, compressOK bool) []byte {
-	shouldCompress := compressOK && len(subject)+len(reply)+len(hdr)+len(msg) > compressThreshold
-
-	elen := 1 + 8 + 8 + len(subject) + len(reply) + len(hdr) + len(msg)
+	// Clip the subject, reply, header and msgs down. Operate on
+	// uint64 lengths to avoid overflowing.
+	slen := min(uint64(len(subject)), math.MaxUint16)
```

Comment:
If we clip that is bad though and we should not try to force through?

Reply:
Maybe, but if we want to enforce sizes, we should be doing that when the message comes into the system, otherwise we could have different behaviour for clustered vs non-clustered setups.

Reply:
What will happen to this message if we clip? Can it be retrieved?

Reply:
Yes, it'll go into the stream as normal and be retrievable. It'll just be truncated. For truncation to happen the subject/reply/headers would need to exceed 64KB and the message would need to exceed 4GB.

Reply:
If we clip the subject then probably not retrievable. But understand that is exceeding 64k.